### ISSUE TITLE
[Security Issue][BUG][IMPORTANT] Improper Input Validation and Numeric Range Comparison Without Minimum Check

### ISSUE THREAD

**Issue by MohamedZidan097** [asim97/SecureCodingDojo_DEBI#18]
**Describe the Security issue **
The instructor send the data through APIs BUT no validation of the data before sending it to the DB
Need to limit the (max_progress) before  sending it to the DB

**Location of the issue**
Path :trainingportal/server.js
File :server.js
Line :633

**Part of the code issue**

``` js script
  //check if the req is from the instructor account
  db.fetchInstructors(null, function(users){  
    for (let i = 0; i < users.length; i++) {
      if (req.user.accountId===users[i].accountId)
      {          
        //update the student infos
        db.getPromise(db.updateStudent,studentUpdates);
        return util.apiResponse(req, res, 200, "Updated Seccesfully");   
      }
    }
    //username not found
    return util.apiResponse(req, res, 400, "Instructor Username NOT FOUND.");
  });
```

**Recommendation**
Use a validator to check the data sent from the instructor and limit the inputs

**example to solve this issue**

``` js script
  if(validator.matches(variable.property,/^[a-zA-Z\s]+$/i)==false){
     return res.send("Invalid Request.");
  }
```

**References**
Common Weakness Enumeration: CWE-20.
Common Weakness Enumeration: CWE-839.

**Comment by asim97** [comment]
This will be our highest perority now....

**Comment by asim97** [comment]
fixed in the follwing commit 
[Security Issue] fix 1  => 64fc3b9
(Issue Closed)
